sync3/extensions/extensions.go

```diff
@@ -202,6 +226,24 @@ func (r Request) ApplyDelta(next *Request) Request {
 	return r
 }
 
+func (r *Request) InterpretAsInitial() {
```

Comment:
I don't know what this is supposed to do.

Reply:
See the comment above the declaration of this in the interface. https://github.com/matrix-org/sliding-sync/pull/149/files#r1232127907

To illustrate: suppose I receive a request whose account data extension is configured with

```json
{
   "enabled": "true"
}
```

If this is a brand-new request, the proxy should process this in two steps:

1. Use the initial sticky parameters for `lists` and `rooms` (i.e. `["*"]`).
2. Set `enabled` to `true`

If this is a follow-up to a previous request, the proxy should interpret this as follows

1. Use the _previous_ sticky values for `lists` and `rooms`.
2. Set `enabled` to `true`.

The way I wanted to handle this was something like:

- On GenericRequest, define a Default() method that returns a new GenericRequest _of the same type_, filled with initial sticky values.
- Use this in Request.ApplyDelta to generate `curr` in the brand-new case when `isNil(curr)`. Then call `curr.ApplyDelta(next)`, just like we already do in the follow-up case.

The problem is that I couldn't figure out a way to do the first bullet. It felt like I was fighting the language. So instead, `InterpretAsInitial` has some extra logic to mutate `next` in place. The intention is that 
`Default().ApplyDelta(next)` would be equal to `next` after running `next.InterpretAsInitial()`.

Reply:
FTAOD I'm not very happy with InterpretAsInitial. If there is some nice way to implement Default() or similar I'm all ears.